src/sql/src/plan/statement/ddl.rs

```diff
@@ -658,8 +699,50 @@ pub fn plan_create_source(
                 // column and casts it to the appropriate target type
                 let mut column_casts = vec![];
                 for (i, column) in table.columns.iter().enumerate() {
-                    let ty = mz_pgrepr::Type::from_oid_and_typmod(column.type_oid, column.type_mod)
-                        .map_err(|e| sql_err!("{}", e))?;
+                    let ty = match text_cols.get(&Oid(table.oid)) {
+                        // Treat the column as text if it was referenced in
+                        // `TEXT COLUMNS`. This is the only place we need to
+                        // perform this logic; even if the type is unsupported,
+                        // we'll be able to ingest its values as text in
+                        // storage.
+                        Some(names) if names.contains(&column.name) => mz_pgrepr::Type::Text,
+                        _ => {
+                            match mz_pgrepr::Type::from_oid_and_typmod(
+                                column.type_oid,
+                                column.type_mod,
+                            ) {
+                                Ok(t) => t,
+                                // If this reference survived purification, we
+                                // do not expect it to be from a table that the
+                                // user will consume., i.e. expect this table to
+                                // be filtered out of table casts.
+                                Err(_) => {
+                                    column_casts.push(
+                                        HirScalarExpr::CallVariadic {
+                                            func: mz_expr::VariadicFunc::ErrorIfNull,
+                                            exprs: vec![
+                                                HirScalarExpr::literal_null(ScalarType::String),
+                                                HirScalarExpr::literal(
+                                                    mz_repr::Datum::from(
+                                                        format!(
+                                                            "Unsupported type with OID {}",
+                                                            column.type_oid
+                                                        )
+                                                        .as_str(),
+                                                    ),
+                                                    ScalarType::String,
+                                                ),
+                                            ],
+                                        }
+                                        .lower_uncorrelated()
+                                        .expect("no correlation"),
+                                    );
+                                    continue;
```

Comment:
```suggestion
```

Reply:
This continue is required because we are not returning an `mz_pgrepr::Type::Text` from this match arm